src/plugins/metrics/prometheus/collector.ts

```diff
@@ -0,0 +1,107 @@
+/**********************************************************************************
+ * MIT License                                                                    *
+ *                                                                                *
+ * Copyright (c) 2021 Hyperjump Technology                                        *
+ *                                                                                *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy   *
+ * of this software and associated documentation files (the "Software"), to deal  *
+ * in the Software without restriction, including without limitation the rights   *
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      *
+ * copies of the Software, and to permit persons to whom the Software is          *
+ * furnished to do so, subject to the following conditions:                       *
+ *                                                                                *
+ * The above copyright notice and this permission notice shall be included in all *
+ * copies or substantial portions of the Software.                                *
+ *                                                                                *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     *
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       *
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    *
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         *
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  *
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  *
+ * SOFTWARE.                                                                      *
+ **********************************************************************************/
+
+import { RequestConfig } from '../../../interfaces/request'
+import { Gauge, register } from 'prom-client'
+import { snakeCase } from 'snake-case'
+import { Probe } from '../../../interfaces/probe'
+
+type requestMetricCollector = {
+  probeID: string
+  requestIndex: number
+  httpStatusCodeGauge: Gauge<string>
+  responseTimeGauge: Gauge<'url' | 'method' | 'statusCode'>
+  responseSizeGauge: Gauge<'url' | 'method' | 'statusCode'>
+}
+
+export class PrometheusCollector {
+  private requestMetricCollectors: requestMetricCollector[] = []
+
+  registerCollectorFromProbes(probes: Probe[]) {
+    // remove all registered metrics
+    register.clear()
+    this.requestMetricCollectors = []
+
+    probes.forEach((probe) => {
+      const { id, requests } = probe
+
+      requests?.forEach((_: RequestConfig, requestIndex: number) => {
+        const metricNamePrefix = `${snakeCase(id)}_request_${requestIndex}`
```

Comment:
This wouldn't work with IDs that uses UUID. Probably we could add handler if the ID is in UUID format, get the first part of the UUID?

Let's say the UUID is like "A-B-C-D" format. We could take the A only.

Reply:
Yeah, thank you for checking. I fixed it here [5da3ac9](https://github.com/hyperjumptech/monika/pull/270/commits/5da3ac91bd6e0c00df4ab4434fa3ca6ace2e71ae).